### DISCUSSION TITLE
How to properly implement a game in a bot?

### DISCUSSION THREAD

**Discussion by DimaDDM** [eternnoir/pyTelegramBotAPI#2022] [Q&A]
Please answer these questions before submitting your issue. Thanks!
Hello. I'm trying to embed an HTML 5 game into a bot, but I'm getting an error when I try to create a keyboard. Please tell me how to correctly send the game to the user by command, for example /game, in the bot. Unfortunately, I could not find examples with this library. Thank you

```
keyboard = telet.InlineKeyboardMarkup()
keyboard.row(telet.InlineQueryResultGame(id=str(uuid.uuid4()), game_short_name="Game Name"))

```
1. What version of pyTelegramBotAPI are you using?
pyTelegramBotApi 4.12.0

2. What OS are you using?
Any OS

3. What version of python are you using?
Python 3.10

**Comment by Badiboy** [comment]
InlineQueryResultGame - is not a key type.

**Reply by DimaDDM** [reply]
Can you provide minimal example, please? Thank you

**Reply by Badiboy** [reply]
I can't.

**Comment by arker8** [comment]
To properly implement a game in a bot, such as a chatbot or Discord bot, you start by conceptualizing the game idea and defining its objectives and interaction mechanics. This involves deciding on the game type (like a quiz, text adventure, or puzzle) and how players will interact with it, usually through text commands. Next, [choose](https://mypcgames.net/gta-vice-city-pc-game-download/) a platform that supports your game's needs, ensuring it has a bot-friendly API. Design the game mechanics carefully, considering the limitations and capabilities of text-based interactions. Then, develop the game using a suitable programming language, integrating with the bot's API to handle user inputs and responses. Testing is crucial to refine gameplay, fix bugs, and improve user experience. Lastly, deploy your game bot, keeping an eye on user feedback for future improvements. This process combines creative game design with technical programming skills, tailored to the unique environment of bot interfaces.
